Test_Engine/Query/VersioningAdapterConstructorList.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2024, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Reflection;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+
+namespace BH.Engine.Test
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Ensures all assemblies are loaded and returns a list of AdapterConstructors to be included when generating a Versioning test set.")]
+        [Input("includeRevit", "Toggle to control wheter assemblies depending on revit API dlls should be loaded or not.")]
+        [MultiOutput(0, "included", "AdapterConstructors to be considered for adding to the Versioning TestSet.")]
+        [MultiOutput(1, "ignored", "AdapterConstructors to _not_ be added to the Versioning TestSet.")]
+        public static Output<List<ConstructorInfo>, List<ConstructorInfo>> VersioningAdapterConstructorList(bool includeRevit = true)
```

Comment:
Do you have a strong rationale behind defaulting `includeRevit` to `true`? I am 50/50 on that tbf, the more I think about it the more I would just leave it without a default value... but no strong opinion, more of intuition

Reply:
Answering here, as answer to both this and the above.

For the current use case, I think this will always be true - we will always want to include everything. I put the option here to allow for some flexibility, that I think we dont really want or need, but thought, might as well - if nothing else to have a clean way to retrigger it.

As to if we should continue if it fails - I do not think so actually. If you asked for it and it fails, then there is honestly no real point in keep on running, as it just means waiting for other things to finish, but you still ending up with incomplete data. Honestly, rather have it just raise error (which is done by the LoadRevitAssemblies method) and exit out with empty lists.